File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcGroup/IfcSystem/IfcStructuralAnalysisModel/IfcStructuralAnalysisModel.cs
public class IfcStructuralAnalysisModel : IfcSystem
{
    public IfcAnalysisModelTypeEnum PredefinedType { get; set; }
    public IfcAxis2Placement3D OrientationOf2DPlane { get; set; }
    public SET [1:?] OF IfcStructuralLoadGroup LoadedBy { get; set; }
    public SET [1:?] OF IfcStructuralResultGroup HasResults { get; set; }
    public IfcObjectPlacement SharedPlacement { get; set; }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : (PredefinedType <> IfcAnalysisModelTypeEnum.USERDEFINED) OR ((PredefinedType = IfcAnalysisModelTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
}

File: src/main/resources/output/IFC4X3_js/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcGroup/IfcSystem/IfcStructuralAnalysisModel/IfcStructuralAnalysisModel.js
class IfcStructuralAnalysisModel extends IfcSystem {
    constructor() {
        /** @type {IFCANALYSISMODELTYPEENUM} */
        this.PredefinedType = null;
        /** @type {IFCAXIS2PLACEMENT3D} */
        this.OrientationOf2DPlane = null;
        /** @type {SET [1:?] OF IFCSTRUCTURALLOADGROUP} */
        this.LoadedBy = null;
        /** @type {SET [1:?] OF IFCSTRUCTURALRESULTGROUP} */
        this.HasResults = null;
        /** @type {IFCOBJECTPLACEMENT} */
        this.SharedPlacement = null;
    }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : (PredefinedType <> IfcAnalysisModelTypeEnum.USERDEFINED) OR ((PredefinedType = IfcAnalysisModelTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
}
